stubs/setuptools/setuptools/build_meta.pyi

```diff
@@ -14,20 +14,37 @@ class Distribution(dist.Distribution):
 
 class _BuildMetaBackend:
     def run_setup(self, setup_script: str = ...) -> None: ...
-    def get_requires_for_build_wheel(self, config_settings: Incomplete | None = ...): ...
-    def get_requires_for_build_sdist(self, config_settings: Incomplete | None = ...): ...
-    def prepare_metadata_for_build_wheel(self, metadata_directory, config_settings: Incomplete | None = ...): ...
+    def get_requires_for_build_wheel(self, config_settings: Mapping[str, Any] | None = ...) -> list[str]: ...
+    def get_requires_for_build_sdist(self, config_settings: Mapping[str, Any] | None = ...) -> list[str]: ...
+    def prepare_metadata_for_build_wheel(
+        self, metadata_directory: str, config_settings: Mapping[str, Any] | None = ...
+    ) -> str: ...
     def build_wheel(
-        self, wheel_directory, config_settings: Incomplete | None = ..., metadata_directory: Incomplete | None = ...
-    ): ...
-    def build_sdist(self, sdist_directory, config_settings: Incomplete | None = ...): ...
+        self, wheel_directory: str, config_settings: Mapping[str, Any] | None = ..., metadata_directory: str | None = ...
+    ) -> str: ...
+    def build_sdist(self, sdist_directory: str, config_settings: Mapping[str, Any] | None = ...) -> str: ...
+    def build_editable(
+        self, wheel_directory: str, config_settings: Mapping[str, Any] | None = ..., metadata_directory: str | None = ...
+    ) -> str: ...
+    def get_requires_for_build_editable(self, config_settings: Mapping[str, Any] | None = ...) -> list[str]: ...
+    def prepare_metadata_for_build_editable(
+        self, metadata_directory: str, config_settings: Mapping[str, Any] | None = ...
+    ) -> str: ...
 
 class _BuildMetaLegacyBackend(_BuildMetaBackend):
     def run_setup(self, setup_script: str = ...) -> None: ...
 
-get_requires_for_build_wheel: Any
-get_requires_for_build_sdist: Any
-prepare_metadata_for_build_wheel: Any
-build_wheel: Any
-build_sdist: Any
-__legacy__: Any
+def get_requires_for_build_wheel(config_settings: Mapping[str, Any] | None = ...) -> list[str]: ...
+def get_requires_for_build_sdist(config_settings: Mapping[str, Any] | None = ...) -> list[str]: ...
+def prepare_metadata_for_build_wheel(metadata_directory: str, config_settings: Mapping[str, Any] | None = ...) -> str: ...
+def build_wheel(
+    wheel_directory: str, config_settings: Mapping[str, Any] | None = ..., metadata_directory: str | None = ...
+) -> str: ...
+def build_sdist(sdist_directory: str, config_settings: Mapping[str, Any] | None = ...) -> str: ...
+def build_editable(
+    wheel_directory: str, config_settings: Mapping[str, Any] | None = ..., metadata_directory: str | None = ...
+) -> str: ...
+def get_requires_for_build_editable(config_settings: Mapping[str, Any] | None = ...) -> list[str]: ...
+def prepare_metadata_for_build_editable(metadata_directory: str, config_settings: Mapping[str, Any] | None = ...) -> str: ...
```

Comment:
We can actually just mimic the runtime more closely here, to reduce code duplication:

```suggestion
_BACKEND: _BuildMetaBackend = ...
get_requires_for_build_wheel = _BACKEND.get_requires_for_build_wheel
get_requires_for_build_sdist = _BACKEND.get_requires_for_build_sdist
prepare_metadata_for_build_wheel = _BACKEND.prepare_metadata_for_build_wheel
build_wheel = _BACKEND.build_wheel
build_sdist = _BACKEND.build_sdist
```

Reply:
Nice, wasn't sure that would work.

Reply:
Also, that's private, which is why I didn't add it originally.

Reply:
> Why `_BACKEND: _BuildMetaBackend = ...` instead of `_BACKEND: _BuildMetaBackend`?

It's a little hack to appease flake8 -- pyflakes doesn't fully support `.pyi` files, and will erroneously complain about `_BuildMetaBackend` being undefined in the assignments that immediately follow the definition of `_BACKEND` if we just do `_BACKEND: _BuildMetaBackend` 🙂

Reply:
> Also, that's private, which is why I didn't add it originally.

Yeah, our policy is basically "Don't include private stuff for the sake of including private stuff, but if it's necessary to improve the typing of public stuff, it's fine" (other exceptions also exist).